trader/trader.go

```diff
@@ -119,24 +122,23 @@ func (t *Trader) Start() {
 		lastUpdateTime       time.Time
 		lastMetricUpdateTime time.Time
 	)
-	startTime := time.Now()
 
 	for {
 		currentUpdateTime := time.Now()
 		if lastUpdateTime.IsZero() || t.timeController.ShouldUpdate(lastUpdateTime, currentUpdateTime) {
 			success := t.update()
-			if shouldSendUpdateMetric(startTime, currentUpdateTime, lastMetricUpdateTime) {
+			if shouldSendUpdateMetric(t.start, currentUpdateTime, lastMetricUpdateTime) {
 				millisForUpdate := time.Since(currentUpdateTime).Milliseconds()
 				e := t.threadTracker.TriggerGoroutine(func(inputs []interface{}) {
 					e := t.metricsTracker.SendUpdateEvent(currentUpdateTime, success, millisForUpdate)
 					if e != nil {
 						log.Printf("failed to send update event metric: %s", e)
 					}
+					lastMetricUpdateTime = currentUpdateTime
```

Comment:
Not sure where my comment went — yes, let’s please add the last metric updated time for send update on the metrics tracker. Will avoid all potential concurrency issues